src/ngraph/op/fused/shuffle_channels.cpp

```diff
@@ -0,0 +1,107 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/op/fused/shuffle_channels.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/op/util/reshape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::ShuffleChannels::ShuffleChannels(const shared_ptr<Node>& data,
+                                     const int axis,
+                                     const size_t groups)
+    : FusedOp("ShuffleChannels", {data})
+    , m_groups{groups}
+{
+    if (axis < 0)
+    {
+        m_axis = axis + data->get_shape().size();
+    }
+    else
+    {
+        m_axis = axis;
+    }
```

Comment:
Move this to pre_validate_and_infer_types. We cannot assume here that 'data' will have a static shape.Can we save axis specification (including negative value) to m_axis and adjust on-the-fly when needed, possibly through a helper method 'get_shuffle_axis'.

Reply:
Ok but in this case:
1. Should I validate the axis value against the input data shape in `pre_validate_and_infer_types` at all? Or should I do it only for static shapes?
2. Where should I adjust the axis with the proposed helper method? Would that be in `decompose_op`?

Reply:
> Ok but in this case:
> 
> 1. Should I validate the axis value against the input data shape in `pre_validate_and_infer_types` at all? Or should I do it only for static shapes?
Right. validate only if input shape is static.
> 2. Where should I adjust the axis with the proposed helper method? Would that be in `decompose_op`?
get_shuffle_axis could be called in decompose_op or from backends when they are figuring out how to execute the op.

Reply:
I've addressed your comment, please have a look again